### PR TITLE
Feature/warnings alert tts speak

### PR DESCRIPTION
**Pull request by vvvakulenko**
Script is a part of TS #666

### CODE REVIEW

**Comment by vvvakulenko** [comment]
@istoimenova  Please review

File: test_scripts/API/Alert/ATF_Alert_TTS_Speak_WARNINGS.lua [hunk 1]
```diff
@@ -0,0 +1,143 @@
+---------------------------------------------------------------------------------------------
```
Comment (vvvakulenko) [inline test_scripts/API/Alert/ATF_Alert_TTS_Speak_WARNINGS.lua]: @istoimenova , updated in 24b4723ac65ec321d63d04282eead0a1d91d840a

Comment (vvvakulenko) [inline test_scripts/API/Alert/ATF_Alert_TTS_Speak_WARNINGS.lua]: @istoimenova , updated in 24b4723ac65ec321d63d04282eead0a1d91d840a

File: test_scripts/API/Alert/ATF_Alert_TTS_Speak_WARNINGS.lua [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+-- Requirement summary:
+-- SDL must send WARNINGS (success:true) to mobile app in case HMI respond WARNINGS at least to one component of RPC
+--
+-- Description:
+-- test is intended to check that SDL sends WARNINGS (success:true) to mobile app in case HMI respond WARNINGS to at least one HMI-portions
+-- in this particular test it is checked case when UI.Alert gets SUCCESS and TTS.Speak gets WARNINGS from HMI 
+-- 1. Used preconditions: App is registered and activated SUCCESSFULLY
+-- 2. Performed steps: 
+-- MOB -> SDL: sends Alert
+-- SDL -> HMI: resends UI.Alert and TTS.Speak to HMI
```
Comment (vvvakulenko) [inline test_scripts/API/Alert/ATF_Alert_TTS_Speak_WARNINGS.lua]: @istoimenova , updated in 24b4723ac65ec321d63d04282eead0a1d91d840a

File: test_scripts/API/Alert/ATF_Alert_TTS_Speak_WARNINGS.lua [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+          self.hmiConnection:SendResponse(data1.id,"BasicCommunication.ActivateApp", "SUCCESS", {})
+        end)
+      end)
+    end
+  end)
+  EXPECT_NOTIFICATION("OnHMIStatus", {hmiLevel = "FULL", systemContext = "MAIN"}) 
+end
+
+function Test.Precondition_PutFile()
+  commonSteps:PutFile("Precondition_PutFile", "icon.png")
```
Comment (vvvakulenko) [inline test_scripts/API/Alert/ATF_Alert_TTS_Speak_WARNINGS.lua]: @istoimenova , updated in 24b4723ac65ec321d63d04282eead0a1d91d840a

File: test_scripts/API/Alert/ATF_Alert_TTS_Speak_WARNINGS.lua [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+end
+
+--[[ Test ]]
+commonFunctions:newTestCasesGroup("Test")
+
+function Test:TestStep_Alert_TTS_Speak_WARNINGS()
+  local cor_id = self.mobileSession:SendRPC("Alert",
```
Comment (vvvakulenko) [inline test_scripts/API/Alert/ATF_Alert_TTS_Speak_WARNINGS.lua]: @istoimenova , updated in 24b4723ac65ec321d63d04282eead0a1d91d840a

File: test_scripts/API/Alert/ATF_Alert_TTS_Speak_WARNINGS.lua [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+    alertType = "BOTH",
+    duration = 3000,
+    progressIndicator = true,
+    softButtons =
+      {{
+        type = "BOTH",
+        text = "Close",
+        image =
+        {
+          value = storagePath.."/icon.png",
```
Comment (vvvakulenko) [inline test_scripts/API/Alert/ATF_Alert_TTS_Speak_WARNINGS.lua]: @istoimenova , updated in 24b4723ac65ec321d63d04282eead0a1d91d840a

File: test_scripts/API/Alert/ATF_Alert_TTS_Speak_WARNINGS.lua [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+          imageType = "DYNAMIC",
+        },
+        isHighlighted = true,
+        softButtonID = 3,
+        systemAction = "DEFAULT_ACTION",
+      }},
+    appID = self.applications["Test Application"]
```
Comment (vvvakulenko) [inline test_scripts/API/Alert/ATF_Alert_TTS_Speak_WARNINGS.lua]: @istoimenova , updated in 24b4723ac65ec321d63d04282eead0a1d91d840a

File: test_scripts/API/Alert/ATF_Alert_TTS_Speak_WARNINGS.lua [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+  })
+  :Do(function(_,data2) self.hmiConnection:SendResponse(data2.id, "UI.Alert", "SUCCESS", { })
```
Comment (vvvakulenko) [inline test_scripts/API/Alert/ATF_Alert_TTS_Speak_WARNINGS.lua]: @istoimenova , updated in 24b4723ac65ec321d63d04282eead0a1d91d840a

File: test_scripts/API/Alert/ATF_Alert_TTS_Speak_WARNINGS.lua [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+  end)
+ 
+  EXPECT_HMICALL("TTS.Speak",
+  {ttsChunks =
+      {{
+        text = "TTSChunk",
+        type = "TEXT"
+    }},
+    speakType = "ALERT",
+    appID = self.applications["Test Application"]
```
Comment (vvvakulenko) [inline test_scripts/API/Alert/ATF_Alert_TTS_Speak_WARNINGS.lua]: @istoimenova , updated in 24b4723ac65ec321d63d04282eead0a1d91d840a

File: test_scripts/API/Alert/ATF_Alert_TTS_Speak_WARNINGS.lua [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+  })
+  :Do(function(_,data3) self.hmiConnection:SendResponse(data3.id, "TTS.Speak", "WARNINGS", { })
```
Comment (vvvakulenko) [inline test_scripts/API/Alert/ATF_Alert_TTS_Speak_WARNINGS.lua]: @istoimenova , updated in 24b4723ac65ec321d63d04282eead0a1d91d840a

**Comment by vvvakulenko** [comment]
@istoimenova  Please review commit 24b4723ac65ec321d63d04282eead0a1d91d840a

File: test_scripts/API/Alert/008_ATF_Alert_TTS_Speak_WARNINGS.lua [hunk 10]
```diff
@@ -0,0 +1,167 @@
...
+          image = { imageType = "DYNAMIC"},
+          isHighlighted = true,
+          softButtonID = 3,
+          systemAction = "DEFAULT_ACTION",
+      }},
+      appID = self.applications[config.application1.registerAppInterfaceParams.appName]
+    })
+    :ValidIf(function(_,data)
+      local value_Icon = storagePath .. "icon.png"
+      if (string.match(data.params.cmdIcon.value, "%S*" .. "("..string.sub(storagePath, 2).."icon.png)" .. "$") == nil ) then
```
Comment (istoimenova) [inline test_scripts/API/Alert/008_ATF_Alert_TTS_Speak_WARNINGS.lua]: @vvvakulenko, the verification should be done for parameter:
data.params.softButtons[1].image.value, not for data.params.cmdIcon.value.
The parameter is taken from array: SoftButtons[], first element of it has structure image{}

File: test_scripts/API/Alert/008_ATF_Alert_TTS_Speak_WARNINGS.lua [hunk 11]
```diff
@@ -0,0 +1,167 @@
...
+        print("\27[31m value of menuIcon is WRONG. Expected: ~".. value_Icon .. "; Real: " .. data.params.cmdIcon.value .. "\27[0m")
```
Comment (istoimenova) [inline test_scripts/API/Alert/008_ATF_Alert_TTS_Speak_WARNINGS.lua]: @vvvakulenko, the verification should be done for parameter:
data.params.softButtons[1].image.value, not for data.params.cmdIcon.value.

File: test_scripts/API/Alert/008_ATF_Alert_TTS_Speak_WARNINGS.lua [hunk 12]
```diff
@@ -0,0 +1,167 @@
...
+  end)
+  EXPECT_NOTIFICATION("OnHMIStatus", {hmiLevel = "FULL", systemContext = "MAIN"})
+end
+
+commonSteps:PutFile("Precondition_PutFile", "icon.png")
+
+--[[ Test ]]
+commonFunctions:newTestCasesGroup("Test")
+
+local resultCodes = {"SUCCESS", "WARNINGS", "WRONG_LANGUAGE", "RETRY", "SAVED"}
```
Comment (istoimenova) [inline test_scripts/API/Alert/008_ATF_Alert_TTS_Speak_WARNINGS.lua]: @vvvakulenko, 
suggestion: will be more readable if you use one empty row

File: test_scripts/API/Alert/008_ATF_Alert_TTS_Speak_WARNINGS.lua [hunk 13]
```diff
@@ -0,0 +1,167 @@
...
+      speakType = "ALERT",
+      appID = self.applications[config.application1.registerAppInterfaceParams.appName]
+    })
+    :Do(function(_,data3)
+      self.hmiConnection:SendNotification("TTS.Started",{ })
+      local function ttsSpeakResponse()
+        self.hmiConnection:SendResponse (data3.id, data3.method, "WARNINGS", {})
+      end
+      RUN_AFTER(ttsSpeakResponse, 1000)
+    end)
```
Comment (istoimenova) [inline test_scripts/API/Alert/008_ATF_Alert_TTS_Speak_WARNINGS.lua]: @vvvakulenko, 
suggestion: will be more readable if you use one empty row

File: test_scripts/API/Alert/008_ATF_Alert_TTS_Speak_WARNINGS.lua [hunk 14]
```diff
@@ -0,0 +1,167 @@
...
+-- in this particular test it is checked case when TTS.Speak gets WARNINGS from HMI and UI.Alert gets ANY successfull result code
+--
+-- 1. Used preconditions:
+-- App is registered and activated SUCCESSFULLY
+-- 2. Performed steps:
+-- MOB -> SDL: sends Alert
+-- HMI -> SDL: TTS.Speak (WARNINGS), UI.Alert (cyclically checked cases for result codes SUCCESS, WARNINGS, WRONG_LANGUAGE, RETRY, SAVED)
+--
+-- Expected result:
+-- SDL -> HMI: resends UI.Alert and TTS.Speak to HM
```
Comment (istoimenova) [inline test_scripts/API/Alert/008_ATF_Alert_TTS_Speak_WARNINGS.lua]: @vvvakulenko, typo mistake: HM, should be HMI

Comment (vvvakulenko) [inline test_scripts/API/Alert/008_ATF_Alert_TTS_Speak_WARNINGS.lua]: @istoimenova , updated in commit f147ec9e2ba145c553fb30b758b2457ac5f51594

File: test_scripts/API/Alert/008_ATF_Alert_TTS_Speak_WARNINGS.lua (hunk 12 above)
Comment (vvvakulenko) [inline test_scripts/API/Alert/008_ATF_Alert_TTS_Speak_WARNINGS.lua]: @istoimenova , updated in commit f147ec9e2ba145c553fb30b758b2457ac5f51594

File: test_scripts/API/Alert/008_ATF_Alert_TTS_Speak_WARNINGS.lua (hunk 10 above)
Comment (vvvakulenko) [inline test_scripts/API/Alert/008_ATF_Alert_TTS_Speak_WARNINGS.lua]: @istoimenova , updated in commit f147ec9e2ba145c553fb30b758b2457ac5f51594

File: test_scripts/API/Alert/008_ATF_Alert_TTS_Speak_WARNINGS.lua (hunk 13 above)
Comment (vvvakulenko) [inline test_scripts/API/Alert/008_ATF_Alert_TTS_Speak_WARNINGS.lua]: @istoimenova , updated in commit f147ec9e2ba145c553fb30b758b2457ac5f51594

**Pull request merged**